ludwig/data/preprocessing.py

```diff
@@ -641,13 +710,12 @@ def preprocess_for_prediction(
         model_definition['preprocessing']
     )
 
-    # Check if hdf5 and json already exist
+    # Check if hdf5 file already exists
     if data_csv is not None:
-        data_hdf5_fp = os.path.splitext(data_csv)[0] + '.hdf5'
+        data_hdf5_fp = get_filepath_in_diff_format(data_csv, 'hdf5')
```

Comment:
maybe call it replace_file_extension to make it more clear what it does?
We can also declare string constants for `csv` and `hdf5`.

Reply:
yeah, was thinking about the string constants. Will add that.